h2/src/main/org/h2/expression/function/Function.java

```diff
@@ -2291,6 +2300,28 @@ private static Value oraHash(Value value, long bucket, long seed) {
         return ValueBigint.get((hc & Long.MAX_VALUE) % (bucket + 1));
     }
 
+    private static Value base64encode(byte[] data, String algorithm) {
+        final String result;
+        if (algorithm == null || algorithm.isEmpty() || "BASE64".equalsIgnoreCase(algorithm)) {
+            result = Base64.getEncoder().encodeToString(data);
+        } else if ("URL".equalsIgnoreCase(algorithm)) {
+            result = Base64.getUrlEncoder().encodeToString(data);
+        } else {
+            throw DbException.getInvalidValueException("Base64 algorithm, valid value: \"BASE64\" (or not set), \"URL\"", algorithm);
+        }
+        return ValueVarchar.get(result);
+    }
+
+    private static Value base64decode(String data) {
+        final byte[] result;
+        if (data.indexOf('_') >= 0 || data.indexOf('-') >= 0) {
+            result = Base64.getUrlDecoder().decode(data);
+        } else {
+            result = Base64.getDecoder().decode(data);
+        }
```

Comment:
You still need to catch a possible exception here and add a test case with invalid input for it.
https://github.com/h2database/h2database/pull/2423#discussion_r377430072